iree/compiler/Codegen/Common/ConvertToDestinationPassingStylePass.cpp

```diff
@@ -304,10 +310,86 @@ static LogicalResult duplicateInitTensorOps(OpBuilder &b,
   return success();
 }
 
+namespace {
+SmallVector<NamedAttribute> PruneAttributeList(linalg::GenericOp op) {
+  auto opAttributes = op.getAttributeNames();
+  llvm::StringSet<> elidedAttrs;
+  elidedAttrs.insert(opAttributes.begin(), opAttributes.end());
+  SmallVector<NamedAttribute> preservedAttrs;
+  for (auto attr : op->getAttrs()) {
+    if (elidedAttrs.count(attr.getName())) continue;
+    preservedAttrs.push_back(attr);
+  }
+  return preservedAttrs;
+}
+
+/// Adapts Linalg ops input operand to output operand. This is required for not
+/// creating extra alloca ops. For more details, see
+/// https://github.com/google/iree/issues/8303
+struct AdaptLinalgInputOperandToOutputOperand
+    : public OpRewritePattern<linalg::GenericOp> {
+  using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::GenericOp op,
+                                PatternRewriter &rewriter) const override {
+    // All the loops should be parallel loops.
+    if (op.getNumLoops() != op.getNumParallelLoops()) return failure();
+    // There is only one result tensor.
+    if (op->getNumResults() != 1) return failure();
```

Comment:
Note: Shouldnt be too hard to generalize this to the case where the number of results is not one. Ok to leave it as is.